lark/lark.py

```diff
@@ -308,28 +309,30 @@ def __init__(self, grammar: 'Union[Grammar, str, IO[str]]', **options) -> None:
                     if self.options.cache is not True:
                         raise ConfigurationError("cache argument must be bool or str")
                         
-                    cache_fn = tempfile.gettempdir() + '/.lark_cache_%s_%s_%s.tmp' % (cache_md5, *sys.version_info[:2])
+                    cache_fn = tempfile.gettempdir() + "/.lark_cache_%s_%s_%s_%s.tmp" % (getpass.getuser(), cache_md5, *sys.version_info[:2])
```

Comment:
I am not sure what the gurantees are about the value returned from `getpass.getuser()`. It could potentially return a string that results in an invalid file name, but we could also wait till someone actually reports a problem with this. In a similar vein, `getuser` might fail on some weird OSes and raise an exception.

Reply:
* Per https://docs.python.org/3/library/getpass.html#getpass.getuser it can raise an exception if attempts at finding a username fail (the exact exception may change in the future: https://github.com/python/cpython/pull/29739) I could wrap `getuser` in a try/except block if that would be preferable?
* As to the invalid filename, I suppose it could be a possibility. On all operating systems I'm aware of, users generally have directories that include the full username.
* Alternatively, perhaps these cached files don't belong in `/tmp` (as in `gettempdir`) but rather the OS-specific user-cache directory (xdg or otherwise, using [appdirs](https://github.com/ActiveState/appdirs)'s `user_cache_dir` for example )

Reply:
Whether anything should be done here depends on whether or not this might fail on some platforms and how often this happens. I would say keep it like this, and we take another look if someone reports this breaking caching for them. But we should keep this in the back of our minds.